pkg/aws/manager_ec2_route_table.go

```diff
@@ -0,0 +1,104 @@
+package aws
+
+import (
+	"fmt"
+	"strings"
+
+	"github.com/aws/aws-sdk-go/aws/awserr"
+
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/aws/session"
+	"github.com/aws/aws-sdk-go/service/ec2"
+	"github.com/aws/aws-sdk-go/service/resourcegroupstaggingapi"
+	"github.com/integr8ly/cluster-service/pkg/clusterservice"
+	"github.com/integr8ly/cluster-service/pkg/errors"
+	"github.com/sirupsen/logrus"
+)
+
+const (
+	loggingKeyRouteTable = "route-table-id"
+
+	resourceTypeRouteTable = "ec2:route-table"
+)
+
+var _ ClusterResourceManager = &RouteTableManager{}
+
+type RouteTableManager struct {
+	ec2Client     ec2Client
+	taggingClient taggingClient
+	logger        *logrus.Entry
+}
+
+func NewDefaultRouteTableManager(session *session.Session, logger *logrus.Entry) *RouteTableManager {
+	return &RouteTableManager{
+		ec2Client:     ec2.New(session),
+		taggingClient: resourcegroupstaggingapi.New(session),
+		logger:        logger.WithField(loggingKeyManager, managerRouteTable),
+	}
+}
+
+func (r *RouteTableManager) GetName() string {
+	return "AWS EC2 RouteTable Manager"
+}
+
+func (r *RouteTableManager) DeleteResourcesForCluster(clusterId string, tags map[string]string, dryRun bool) ([]*clusterservice.ReportItem, error) {
+	r.logger.Debug("delete route table resources for cluster")
+	resourceInput := &resourcegroupstaggingapi.GetResourcesInput{
+		ResourceTypeFilters: aws.StringSlice([]string{resourceTypeRouteTable}),
+		TagFilters:          convertClusterTagsToAWSTagFilter(clusterId, tags),
+	}
+	resourceOutput, err := r.taggingClient.GetResources(resourceInput)
+	if err != nil {
+		return nil, errors.WrapLog(err, "failed to filter route tables", r.logger)
+	}
+	var routeTablesToDelete []*basicResource
+	for _, resourceTagMapping := range resourceOutput.ResourceTagMappingList {
+		arn := aws.StringValue(resourceTagMapping.ResourceARN)
+		arnElements := strings.Split(arn, ":")
+		routeTableId := arnElements[len(arnElements)-1]
+		if routeTableId == "" {
+			return nil, errors.WrapLog(err, fmt.Sprintf("invalid route table name from arn, %s", routeTableId), r.logger)
+		}
```

Comment:
Would it be overkill to have a little wrapper function for this? It seems to be used quite a bit across all of the files

Reply:
maybe but we found that the split is not always the same sometime is `/` and others its `:`  so may not be much benefit